contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -2775,9 +2776,18 @@ TdsSendTypeVarchar(FmgrInfo *finfo, Datum value, void *vMetaData)
 								 * store given string in given encoding. */
 				maxLen;			/* max size of given column in bytes */
 	char	   *destBuf,
-			   *buf = OutputFunctionCall(finfo, value);
+			   *buf;
 	TdsColumnMetaData *col = (TdsColumnMetaData *) vMetaData;
 
+	/*
+	* If client being connected is using TDS version lower than or equal to
+	* 7.1 then TSQL treats varchar(max) as Text.
+	*/
+	if (GetClientTDSVersion() <= TDS_VERSION_7_1_1 && 
+			col->metaEntry.type3.tdsTypeId == TDS_TYPE_TEXT)
+		return TdsSendTypeText(finfo, value, vMetaData);
+
```

Comment:
If we're handling it here, why do we need to handle in PrepareRowDescription as well? Also, this is not setting sendTableName to true although we're sending as text.